aesara/tensor/nnet/abstract_conv.py

```diff
@@ -2543,18 +2508,23 @@ def make_node(self, img, kern):
             if kern.type.ndim != 2 + self.convdim:
                 raise TypeError(f"kern must be {int(2 + self.convdim)}D tensor")
 
-        img = assert_shape(
-            img,
-            self.imshp,
-            "AbstractConv shape mismatch: shape of "
-            "image does not match given imshp.",
-        )
-        kern = assert_shape(
-            kern,
-            self.kshp,
-            "AbstractConv shape mismatch: shape of "
-            "filters does not match given kshp.",
-        )
+        img_shape = img.shape
+        tests = []
+        for i in range(img.ndim):
+            if self.imshp[i] is not None:
+                tests.append(aet.eq(img_shape[i], self.imshp[i]))
+        if tests:
+            raise AssertionError(
+                "AbstractConv shape mismatch: shape of image does not match given imshp.")
```

Comment:
We can't replace the functionality of `Assert` like this, because `Assert` is an `Op` that performs the check at "run-time" (i.e. when the graph is being evaluated).

Instead, we should try replacing `assert_shape` with `SpecifyShape`.

Reply:
I apologize for the delay in replying, but I looked through the SpecifyShape class, and how it's being used across the codebase. However, I'm not being able to understand how I can achieve the functionality of assert_shape using SpecifyShape. Here's my current approach (following the method used in `tests.tensor.nnet.test_abstract_conv`):

```python
specify_shape = SpecifyShape()
image_func = aesara.function([img], specify_shape(img, [self.imshp]))
kern_func = aesara.function([kern], specify_shape(kern, [self.kshp]))
image_func(img)
kern_func(kern)
```
However, on running the tests, the AssertionError is being produced at a stage before it is expected, and hence the tests will fail. Is this the right way to approach this? If not, please do suggest how I can go about doing this.

Reply:
It's better if you use the existing instance of `SpecifyShape` in `aesara.tensor.shape`.

Otherwise, you should only need to replace `assert_shape` with calls to `specify_shape`.  If there are errors caused by that, you'll need to rebase this branch with those changes so I can see them.

Reply:
I've replaced the calls to assert_shape with specify_shape. However, the unit tests in `tests.tensor.nnet.test_abstract_conv` will fail due to TypeErrors.